modules/fpga.c

```diff
@@ -54,10 +54,10 @@ STATIC MP_DEFINE_CONST_FUN_OBJ_2(fpga_read_obj, fpga_read);
 
 STATIC mp_obj_t fpga_write(mp_obj_t addr_16bit, mp_obj_t bytes)
 {
-    size_t n;
-    const char *buffer = mp_obj_str_get_data(bytes, &n);
+    mp_buffer_info_t buffer;
+    mp_get_buffer_raise(bytes, &buffer, MP_BUFFER_READ);
```

Comment:
This change seemed to be needed to be able to pass a `bytearray()` in addition to a string-like object such as `b'\x00\x01\x02'`

Reply:
cool!